libs/community/langchain_community/chat_message_histories/redis.py

```diff
@@ -57,6 +59,8 @@ def messages(self) -> List[BaseMessage]:  # type: ignore
     def add_message(self, message: BaseMessage) -> None:
         """Append the message to the record in Redis"""
         self.redis_client.lpush(self.key, json.dumps(message_to_dict(message)))
+        if self.history_size:
+            self.redis_client.ltrim(self.key, 0, self.history_size - 1)
```

Comment:
Does this remove the messages in the db? My first thought would be to not delete messages but just fetch a window based on `history_size`.